robocop/checkers/misc.py

```diff
@@ -134,3 +127,86 @@ def visit_KeywordCall(self, node):  # noqa
                     col = token.col_offset + 1
                     break
             self.report("if-can-be-used", node.keyword, node=node, col=col)
+
+
+class ConsistentAssignmentSignChecker(VisitorChecker):
+    """ Checker for inconsistent assignment signs.
+
+    By default this checker will try to autodetect most common assignment sign (separately for *** Variables *** section
+    and (*** Test Cases ***, *** Keywords ***) sections and report any not consistent type of sign in particular file.
+
+    To force one type of sign type (to emulate now deprecated ``0906 (redundant-equal-sign)`` rule) you can configure
+    two rules::
+
+        --configure inconsistent-assignment-sign:assignment_sign_type:{sign_type}
+        --configure inconsistent-assignment-sign-variables:assignment_sign_type:{sign_type}
+
+    ``${sign_type}` can be one of: ``autodetect`` (default), ``equal_sign`` ('='), ``none`` (''),
+    ``space_and_equal_sign`` (' =').
+
+    """
+    rules = {
+        "0909": (
+            "inconsistent-assignment-sign",
```

Comment:
I think I would just name it `inconsistent-assignment`

Reply:
Sweet, a lot of names will get shorter that way :D